frappe/templates/includes/comments/comments.py

```diff
@@ -9,7 +9,7 @@
 
 from frappe import _
 
-@frappe.whitelist()
+@frappe.whitelist(allow_guest=True)
 def add_comment(comment, comment_email, comment_by, reference_doctype, reference_name, route):
```

Comment:
Any user (guest) can still comment using REST-API irrespective of client-side validation... 
you have to validate comment access server-side as well.